demisto_sdk/commands/common/constants.py

```diff
@@ -915,25 +922,45 @@ def __init__(self, repo_name: Optional[str] = None):
         else:
             self.CURRENT_REPOSITORY = repo_name
         # DO NOT USE os.path.join on URLs, it may cause errors
-        self.CONTENT_GITHUB_LINK = urljoin(self.BASE_RAW_GITHUB_LINK, self.CURRENT_REPOSITORY)
-        self.CONTENT_GITHUB_MASTER_LINK = urljoin(self.CONTENT_GITHUB_LINK, r'master')
-        self.Credentials = GithubCredentials()
+        if not self.IS_GITLAB:
+            self.CONTENT_GITHUB_LINK = urljoin(self.BASE_RAW_GITHUB_LINK, self.CURRENT_REPOSITORY)
+            self.CONTENT_GITHUB_MASTER_LINK = urljoin(self.CONTENT_GITHUB_LINK, r'master')
+        else:
+            self.BASE_RAW_GITLAB_LINK = self.BASE_RAW_GITLAB_LINK.format(GITLAB_ID=self.GITLAB_ID)
 
-    @staticmethod
-    def _get_repository_name(urls: Iterable) -> str:
+    # @staticmethod
+    def _get_repository_name(self, urls: Iterable) -> str:
         """Returns the git repository of the cwd.
         if not running in a git repository, will return an empty string
         """
         try:
             for url in urls:
-                repo = re.findall(r'.com[/:](.*)', url)[0].replace('.git', '')
-                return repo
+                if 'github' in url:
+                    return re.findall(r'.com[/:](.*)', url)[0].replace('.git', '')
+                else:  # gitlab
+                    repo = re.findall(r'code\.pan\.run/xsoar[/:](.*)', url)[0].replace('.git', '')
+                    self.GITLAB_ID = self._search_gitlab_id(repo)
+                    if self.GITLAB_ID is None:
+                        continue
+                    self.IS_GITLAB = True
+                    return repo
         except (AttributeError, IndexError):
             pass
 
         # default to content repo if the repo is not found
         click.secho('Could not find the repository name - defaulting to demisto/content', fg='yellow')
-        return GithubContentConfig.OFFICIAL_CONTENT_REPO_NAME
+        return GitContentConfig.OFFICIAL_CONTENT_REPO_NAME
+
+    @lru_cache(maxsize=128)
+    def _search_gitlab_id(self, repo: str) -> Optional[int]:
```

Comment:
add a test for this method (mainly to have records of a valid GitLab response example).

Reply:
added tests